File: src/my_cpp_pkg/src/robot_news.cpp
#include <rclcpp/rclcpp.hpp>
#include <example_interfaces/msg/string.hpp>

using namespace std::chrono_literals;
using example_interfaces::msg::String;

class RobotNews : public rclcpp::Node
{
public:
    RobotNews() : Node{"robot_news"}, name_{"R2D2"}
    {
        timer_ = this->create_wall_timer(1s, [this](){
            this->publish_news();
        });
        publisher_ = this->create_publisher<String>("robot_news", 10);
        RCLCPP_INFO(this->get_logger(), "Robot News Started!");
    }

    void publish_news() {
        String msg = String();
        msg.data = "Hello, this is " + name_ + " from Robot News Station";
        publisher_->publish(msg);
    }

private:
    rclcpp::Publisher<String>::SharedPtr publisher_;
    rclcpp::TimerBase::SharedPtr timer_;
    std::string name_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<RobotNews>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: src/my_py_pkg/my_py_pkg/robot_news.py
#/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import String

class RobotNewsNode(Node):
    def __init__(self):
        super().__init__("robot_news")
        self.name_ = "C3P0"
        self.timer_ = self.create_timer(1, self.publish_news)
        self.publisher_ = self.create_publisher(String, "robot_news", 10)
        self.get_logger().info("Robot News Started!")

    def publish_news(self):
        msg = String()
        msg.data = "Hello this is " + self.name_ + " from Robot News Station."
        self.publisher_.publish(msg)


def main(args=None):
    rclpy.init()
    node = RobotNewsNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()